test/integrate-pummel-leak-test.js

```diff
@@ -104,20 +103,21 @@ function runChild () {
 
   function runOperation () {
```

Comment:
some things, like this are much easier with async / await also release may be async and your not awaiting it.

```js
async  function runOperation () {
    const { connection } = await db.getConnection()
    const query = promisify(connection.query)
    const data = await connection.query('SELECT 1')
    await release()
    return data
  }
```